marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/ingestion/bulkIngester.sjs

```diff
@@ -38,16 +61,23 @@ inputs.forEach(record => {
   state.next = state.next + 1;
   const uri = (state.uriprefix) +  sem.uuidString() + '.json';
   record = ingest.main({uri: uri, value: record}, {
-    outputFormat: consts.JSON, headers: {createdOn: consts.CURRENT_DATE_TIME, createdBy: consts.CURRENT_USER}
+    outputFormat: consts.JSON,
+                  headers: {
+                      createdOn: consts.CURRENT_DATE_TIME,
+                      createdBy: consts.CURRENT_USER,
+                      sources: [
+                        { name: work.sourcename,
```

Comment:
We only want to add a source object if at least one of these is set. So you should add a test that verifies that when neither sourcename nor sourcetype is provided, then the sources array does not exist.